Assets/Scripts/Fire.cs

```diff
@@ -1,45 +1,46 @@
 ﻿using System.Collections;
-using System;
-using System.Collections.Generic;
 using UnityEngine;
-using UnityEngine.UI;
-public class Fire : MonoBehaviour
+using UnityEngine.Assertions;
+using Liminal.SDK.VR;
+using Liminal.SDK.VR.Input;
+using Liminal.SDK.VR.Pointers;
+
+/// <summary>
+/// <see cref="Fire"/> Is used to control the player's laser and call the destroy interface on enemy objects via a spherical raycast.
+/// </summary>
+public class Fire 
+    : MonoBehaviour
 {
-    [Header("Events")]
-    [Tooltip("Raised when Button Pressed")]
-    public UnityEngine.Events.UnityEvent OnPressed = new UnityEngine.Events.UnityEvent();
-    #region MonoBehaviour
-    private void Update()
-    {
-        this.gameObject.transform.position = obj.transform.position;
-        this.gameObject.transform.rotation = obj.transform.rotation;
-        var pointer = Liminal.SDK.VR.VRDevice.Device.PrimaryInputDevice.Pointer;
-
-        //this.gameObject.transform.localPosition = obj.transform.localPosition;
-        FireGun();
-    }
-    #endregion
     public GameObject obj;
     public GameObject ExplosionEffect;
     public GameObject FireEffect;
-    public GameObject FireEffectSP;
     public GameObject earth;
-    public GameObject longboi;
-    GameObject lazer;
+    
+    [Header("Laser Details")]
```

Comment:
Keep variables, methods and properties separated.

Reply:
Done